_posts/all-new-goodies-in-servicecomposer.md

```diff
@@ -0,0 +1,179 @@
+---
+layout: post
+title: "All the new goodies in ServiceComposer"
+author: Mauro Servienti
+synopsis: "Thanks to users’ feedback, I enjoyed working on some new cool ServiceComposer features, namely endpoint filters, a greatly improved event handling API, and declarative model binding, which paves the way for a new cool feature in the works"
+header_image: /img/posts/all-new-goodies-in-servicecomposer/header.jpg
+tags:
+- viewmodel-composition
+- development
+---
+
+It's been a while since [the last blog post](https://milestone.topics.it/2024/04/09/service-control-ghost-endpoints.html), and no, the [20th anniversary](https://milestone.topics.it/2024/11/30/anniversary.html) doesn't count.
+
+Thanks to some user requests, I spent some time working on [ServiceComposer](https://github.com/ServiceComposer/ServiceComposer.AspNetCore), my [ViewModel Composition](https://milestone.topics.it/series/view-model-composition.html) open-source library. I ended up releasing two major releases and one minor release for each one of the majors. A lot of work!
+
+In a few weeks, I released [3.0](https://github.com/ServiceComposer/ServiceComposer.AspNetCore/releases/tag/3.0.0), [3.1](https://github.com/ServiceComposer/ServiceComposer.AspNetCore/releases/tag/3.1.0), [4.0](https://github.com/ServiceComposer/ServiceComposer.AspNetCore/releases/tag/4.0.0), and [4.1](https://github.com/ServiceComposer/ServiceComposer.AspNetCore/releases/tag/4.1.0).
+
+## Versions 3.0 and 3.1
+
+Let's look at what's new in ServiceComposer versions [3.0](https://github.com/ServiceComposer/ServiceComposer.AspNetCore/releases/tag/3.0.0) and [3.1](https://github.com/ServiceComposer/ServiceComposer.AspNetCore/releases/tag/3.1.0). I'll refer to those versions as 3.x.
+
+### Endpoint filters
+
+Version 3.x introduced support for [endpoint filters](https://github.com/ServiceComposer/ServiceComposer.AspNetCore/blob/master/docs/endpoint-filters.md) using the standard ASP.Net minimal API filters syntax. This feature allows for more granular control over the request pipeline, enhancing the flexibility and power of ServiceComposer. The decision to use the minimal API syntax and programming model is to avoid introducing yet another programming style. Overall, ServiceComposer is leaning towards being a citizen in the ASP.Net ecosystem. As such, its API should be aligned as much as possible with the Microsoft one to maintain the 'feel at home' feeling. More on this later.
```

Comment:
```suggestion
Version 3.x introduced support for [endpoint filters](https://github.com/ServiceComposer/ServiceComposer.AspNetCore/blob/master/docs/endpoint-filters.md) using the standard ASP.Net minimal API filters syntax. This feature allows more granular control over the request pipeline, enhancing the flexibility and power of ServiceComposer. The decision to use the minimal API syntax and programming model is to avoid introducing yet another programming style. Overall, ServiceComposer is leaning towards being a citizen in the ASP.Net ecosystem. As such, its API should be aligned as much as possible with the Microsoft one to maintain the 'feel at home' feeling. More on this later.
```